File: Napakalaki-Ruby/lib/numeric_bad_consequence.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module NapakalakiGame
class NumericBadConsequence < BadConsequence
  def initialize(text,levels,someVisibleTreasures,someHiddenTreasures)
    super(text,levels)    
    @nVisibleTreasures=someVisibleTreasures
    @nHiddenTreasures=someHiddenTreasures    
  end
  
  def isEmpty
    (@nHiddenTreasures==0 && @nVisibleTreasures==0)
  end
  
  def substractVisibleTreasure(t)
    if(@nVisibleTreasures>0)
      @nVisibleTreasures=@nVisibleTreasures - 1
    end
  end
  
  def substractHiddenTreasure(t)
    if(@nHiddenTreasures>0)
      @nHiddenTreasures = @nHiddenTreasures - 1
    end 
  end
  
  def getNVisibleTreasures
    @nVisibleTreasures
  end
  
  def getNHiddenTreasures
    @nHiddenTreasures
  end
  
  def adjustToFitTreasureLists(v,h)
    a_quitarV = @nVisibleTreasures>v.size() ? v.size() : @nVisibleTreasures
    a_quitarH = @nHiddenTreasures>h.size() ? h.size() : @nHiddenTreasures
    NumericBadConsequence.new(@text,0,a_quitarV,a_quitarH)
  end
  
  def to_s
    out = @text
    out = out + ", pierdes #{@levels} niveles"
    if(@nVisibleTreasures >= 0 && @nHiddenTreasures>=0 )
      out =out + ", pierdes #{@nVisibleTreasures} tesoros visibles y #{@nHiddenTreasures} tesoros ocultos"
    end
    out     
  end
  
end
end

File: Napakalaki-Java/src/NapakalakiGame/NumericBadConsequence.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

import java.util.ArrayList;

/**
 *
 * @author Laura
 */
public class NumericBadConsequence extends BadConsequence {
    private int nVisibleTreasures;
    private int nHiddenTreasures;
    public NumericBadConsequence(String text_i, int levels_i, int nVisible, int nHidden){
        super(text_i, levels_i);
        nVisibleTreasures=nVisible;
        nHiddenTreasures=nHidden;
    }
    
    @Override
    public boolean isEmpty(){
       return (nVisibleTreasures==0 && nHiddenTreasures==0);
    }
    
    @Override
    void substractVisibleTreasure(Treasure t){
        if(nVisibleTreasures!=0)
            nVisibleTreasures--;
    }
    @Override
    void substractHiddenTreasure(Treasure t){
        if(nHiddenTreasures!=0)
            nHiddenTreasures--;
    }
    @Override
    BadConsequence adjustToFitTreasureLists(ArrayList<Treasure> v, ArrayList<Treasure> h){
       int nuevoNV,nuevoNH;
       nuevoNV=nVisibleTreasures>v.size()?v.size():nVisibleTreasures;
       nuevoNH=nHiddenTreasures>h.size()?h.size():nHiddenTreasures; 
       return new NumericBadConsequence(text,levels,nuevoNV,nuevoNH);
    }
    
    public int getNVisibleTreasures(){
        return nVisibleTreasures;
    }
    public int getNHiddenTreasure(){
        return nHiddenTreasures;
    }
    @Override
    public String toString(){
        String out;
        out =  super.toString();
        if(nVisibleTreasures>=0)
            out += ", number of Visible Treasures = " + Integer.toString(nVisibleTreasures);
        if(nHiddenTreasures>=0)
            out += ", number of Hidden Treasures = " + Integer.toString(nHiddenTreasures);
        return out;
    }
}
